go/internal/utils/transform_utils.go

```diff
@@ -101,3 +105,12 @@ func ToString(v any) (string, bool) {
 		return fmt.Sprintf("%v", val), true
 	}
 }
+
+// Convert to and perform bound checks for uint32 representation for milliseconds
+func DurationToMilliseconds(d time.Duration) (uint32, error) {
+	milliseconds := d.Milliseconds()
+	if milliseconds < 0 || milliseconds > math.MaxUint32 {
+		return 0, &errors.ConfigurationError{Msg: "Invalid duration was specified"}
```

Comment:
Go error strings should not be capitalized: https://go.dev/wiki/CodeReviewComments#error-strings